src/solvers/file_solution_history.C

```diff
@@ -259,11 +259,25 @@ void FileSolutionHistory::retrieve(bool is_adjoint_solve, Real time)
 
 void FileSolutionHistory::erase(Real time)
 {
-  stored_solutions_iterator stored_sols_erase_it;
+  // We cant erase the stored_sols iterator which is used in other places
+  // So save its current value for the future
+  stored_solutions_iterator stored_sols_last = stored_sols;
 
-  stored_sols_erase_it = stored_solutions.find(time);
+  // This will map the stored_sols iterator to the current time
+  this->find_stored_entry(time);
+
+  // map::erase behaviour is undefined if the iterator is pointing
+  // to a non-existent element.
+  libmesh_assert(stored_sols != stored_solutions.end());
+
+  // We want to keep using the stored_sols iterator, so we have to create
+  // a new one to erase the concerned entry
+  stored_solutions_iterator stored_sols_copy = stored_sols;
+  stored_sols = stored_sols_last;
+  stored_sols--;
```

Comment:
Surely we don't want to do this every time?  If we're asking to erase the entry at stored_sols, then sure, move stored_sols somewhere safer first, but otherwise this is just a weird unintended side effect.